docs/core/02-how-to-guides/01-plugin-development/01-implemention/01-plugin-types.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Plugin Types
+---
+
+## Choosing the Base Contract for Your Plugin
+
+In this section, we will learn about the interfaces or base contracts to inherit from when developing a plugin.
+
+Depending on the use-case of your plugin, you might want it to be
+
+- upgradeable or non-upgradeable
+- deployed by a specific deployment method
+- compatible with meta transactions
+
+:::caution
+Upgradeable plugin contracts (such as `PluginUpgradeable`, `PluginUUPSUpgradeable` implementations) must reserve storage slots by defining a storage gap. If this is not done properly and/or variables are rearranged in the upgrade, the storage can become corrupted rendering it inaccessible and resulting in the loss of funds.
```

Comment:
Great add-on! I'd say let's add a link here to "storage gap" for someone who doesn't know what that is.

Reply:
Provided link to [OZs storage gap article](https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps).